deepmd/train/trainer.py

```diff
@@ -345,6 +348,9 @@ def _build_training(self):
             optimizer = self.run_opt._HVD.DistributedOptimizer(optimizer)
         else:
             optimizer = tf.train.AdamOptimizer(learning_rate = self.learning_rate)
+        if DP_ENABLE_MIXED_PRECISION:
+            # enable dynamic loss scale of the gradients
+            optimizer = tf.train.experimental.enable_mixed_precision_graph_rewrite(optimizer)
```

Comment:
This function has been moved to `tf.mixed_precision.enable_mixed_precision_graph_rewrite`. https://www.tensorflow.org/api_docs/python/tf/compat/v1/mixed_precision/enable_mixed_precision_graph_rewrite What TF version do you use? Do you know if it is supported in all TF versions?

Reply:
This function was found in Nvidia's official [documentation](https://docs.nvidia.com/deeplearning/performance/mixed-precision-training/index.html#tensorflow-amp). I have tested it with the TF-1.14.0 and TF-2.6.0 environment. Since it is a deprecated function, I will use the new `tf.mixed_precision.enable_mixed_precision_graph_rewrite` function.

Reply:
The method was available since v1.12 (https://github.com/tensorflow/tensorflow/commit/02730dcecf0e479c5be06adab16d5f6a0c03d11c) and then was renamed in v2.4 (https://github.com/tensorflow/tensorflow/commit/011228639301a8ed60d39c94ab096b074091bb4a#diff-38a59d213566539b30a6ce96962d40b68f18db2365e74e5e8bec220c43e92aec).

Reply:
Sure